src/main/java/tira/Parser.java

```diff
@@ -0,0 +1,54 @@
+package tira;
+
+import tira.task.TaskList;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+/*
+Parser class:
+1. Splits the initial command and then calls the appropriate function
+if command is list, calls the getTaskList method from TaskList
+if command is delete, cal
+ */
+public class Parser {
+    private TaskList taskList;
+    private final Ui ui = new Ui();
+
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    public void parseCommand(String command) throws TiraException {
+        String [] splitCommand = command.split(" ");
+        String taskType = splitCommand[0];
+        try {
+            switch (taskType) {
```

Comment:
Shouldn't there be no indentations for `case` clauses?

Reply:
Agreed! Take care of indentations for switch case blocks, intellij IDEA automatically indents the case statements, however this violates the coding style followed by CS2103.